react/features/calendar-sync/middleware.js

```diff
@@ -265,24 +269,39 @@ function _updateCalendarEntries(events) {
     if (events && events.length) {
         // eslint-disable-next-line no-invalid-this
         const { dispatch, getState } = this;
-
         const knownDomains = getState()['features/base/known-domains'];
-        const eventList = [];
-
         const now = Date.now();
+        const entryMap = new Map();
 
         for (const event of events) {
             const calendarEntry = _parseCalendarEntry(event, knownDomains);
 
-            calendarEntry
-                && calendarEntry.endDate > now
-                && eventList.push(calendarEntry);
+            if (calendarEntry && calendarEntry.endDate > now) {
+                // This is the data structure we'll try to match all entries to.
+                // The smaller the better, hence the single letter field names.
+                const eventMatchHash = md5.hex(JSON.stringify({
+                    d: new Date(calendarEntry.startDate).toTimeString(),
+                    t: calendarEntry.title,
+                    u: calendarEntry.url
+                }));
+                const existingEntry = entryMap.get(eventMatchHash);
```

Comment:
So... the startDate is an argument of the hash i.e. it's part of the key?! But isn't it supposed to not be part of the key because we want to show only the earliest startDate of the same URL (and title I guess)? How is the `entryMap.get` going to work given that we expect to check `existingEntry.startDate > event.startDate`? I think it doesn't really work.